### ISSUE TITLE
Upgrading from Stable Elemental Teal to Dev Elemental Teal fails

### ISSUE THREAD

**Issue by davidcassany** [rancher/elemental#632]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

The OS upgrade fails. Mostly because the upgrade hooks are applied with a wrong environment

**What did you expect to happen:**

A successful upgrade

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

The issue is related with the suc-upgrade script and with its runtime environment.

**Environment:**

- Elemental release version (use `cat /etc/os-release`):
- Rancher version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:

**Comment by frelon** [comment]
Update on this issue: there was a problem in the elemental-toolkit package for upgrade_grub_hooks where the mountpoint of COS_STATE was hardcoded, but in suc-upgrade the system is mounted to `/host` . This left the upgraded system unbootable since new GRUB labels were not set.

The fix is implemented in rancher/elemental-cli#414 and we will have the old hook removed in #657

**Comment by frelon** [comment]
Running an update stable->dev now seems to be working again, the system boots and the relevant logs look promising:

```
INFO[2023-02-07T14:08:36Z] Running after-upgrade hook
ERRO[2023-02-07T14:08:37Z] grub2-editenv: error: cannot open `/run/initramfs/cos-state/grub_oem_env': Read-only file system.
grub2-editenv: error: cannot open `/run/initramfs/cos-state/grub_oem_env': Read-only file system.
grub2-editenv: error: cannot open `/run/initramfs/cos-state/grub_oem_env': Read-only file system.
grub2-editenv: error: cannot open `/run/initramfs/cos-state/grub_oem_env': Read-only file system.
grub2-editenv: error: cannot open `/run/initramfs/cos-state/grub_oem_env': Read-only file system.
grub2-editenv: error: cannot open `/run/initramfs/cos-state/grub_oem_env': Read-only file system.
: failed to run statemnt="/run/cos/state"
if [ ! -f "${statemnt}/grub2/grub.cfg" ]; then
statemnt="/run/initramfs/cos-state"
fi
grub2-editenv ${statemnt}/grub_oem_env set state_label=COS_STATE
grub2-editenv ${statemnt}/grub_oem_env set active_label=COS_ACTIVE
grub2-editenv ${statemnt}/grub_oem_env set passive_label=COS_PASSIVE
grub2-editenv ${statemnt}/grub_oem_env set recovery_label=COS_RECOVERY
grub2-editenv ${statemnt}/grub_oem_env set system_label=COS_SYSTEM
grub2-editenv ${statemnt}/grub_oem_env set oem_label=COS_OEM
: exit status 1
ERRO[2023-02-07T14:08:37Z] 1 error occurred:
* failed to run statemnt="/run/cos/state"
if [ ! -f "${statemnt}/grub2/grub.cfg" ]; then
statemnt="/run/initramfs/cos-state"
fi
grub2-editenv ${statemnt}/grub_oem_env set state_label=COS_STATE
grub2-editenv ${statemnt}/grub_oem_env set active_label=COS_ACTIVE
grub2-editenv ${statemnt}/grub_oem_env set passive_label=COS_PASSIVE
grub2-editenv ${statemnt}/grub_oem_env set recovery_label=COS_RECOVERY
grub2-editenv ${statemnt}/grub_oem_env set system_label=COS_SYSTEM
grub2-editenv ${statemnt}/grub_oem_env set oem_label=COS_OEM
: exit status 1
DEBU[2023-02-07T14:08:37Z] Running grub2-editenv with params: /host/run/initramfs/cos-state/grub_oem_env set passive_label=COS_PASSIVE
DEBU[2023-02-07T14:08:37Z] Running cmd: 'grub2-editenv /host/run/initramfs/cos-state/grub_oem_env set passive_label=COS_PASSIVE'
DEBU[2023-02-07T14:08:37Z] Running grub2-editenv with params: /host/run/initramfs/cos-state/grub_oem_env set recovery_label=COS_RECOVERY
DEBU[2023-02-07T14:08:37Z] Running cmd: 'grub2-editenv /host/run/initramfs/cos-state/grub_oem_env set recovery_label=COS_RECOVERY'
DEBU[2023-02-07T14:08:37Z] Running grub2-editenv with params: /host/run/initramfs/cos-state/grub_oem_env set system_label=COS_SYSTEM
DEBU[2023-02-07T14:08:37Z] Running cmd: 'grub2-editenv /host/run/initramfs/cos-state/grub_oem_env set system_label=COS_SYSTEM'
DEBU[2023-02-07T14:08:37Z] Running grub2-editenv with params: /host/run/initramfs/cos-state/grub_oem_env set oem_label=COS_OEM
DEBU[2023-02-07T14:08:37Z] Running cmd: 'grub2-editenv /host/run/initramfs/cos-state/grub_oem_env set oem_label=COS_OEM'
DEBU[2023-02-07T14:08:37Z] Running grub2-editenv with params: /host/run/initramfs/cos-state/grub_oem_env set persistent_label=COS_PERSISTENT
DEBU[2023-02-07T14:08:37Z] Running cmd: 'grub2-editenv /host/run/initramfs/cos-state/grub_oem_env set persistent_label=COS_PERSISTENT'
DEBU[2023-02-07T14:08:37Z] Running grub2-editenv with params: /host/run/initramfs/cos-state/grub_oem_env set state_label=COS_STATE
DEBU[2023-02-07T14:08:37Z] Running cmd: 'grub2-editenv /host/run/initramfs/cos-state/grub_oem_env set state_label=COS_STATE'
DEBU[2023-02-07T14:08:37Z] Running grub2-editenv with params: /host/run/initramfs/cos-state/grub_oem_env set active_label=COS_ACTIVE
DEBU[2023-02-07T14:08:37Z] Running cmd: 'grub2-editenv /host/run/initramfs/cos-state/grub_oem_env set active_label=COS_ACTIVE'
INFO[2023-02-07T14:08:37Z] rebranding
DEBU[2023-02-07T14:08:37Z] Looking for GRUB_ENTRY_NAME name in /run/cos/workingtree/etc/os-release
INFO[2023-02-07T14:08:37Z] Setting default grub entry to Elemental

```
